src/python/pants/engine/subsystem/native.py

```diff
@@ -550,57 +599,81 @@ def visualize_to_dir(self):
     return self._visualize_to_dir
 
   @memoized_property
+  def binary(self):
+    """Load and return the path to the native engine binary."""
+    return self._binary_util.select_binary(self._supportdir,
+                                           self._version,
+                                           '{}.so'.format(NATIVE_ENGINE_MODULE))
+
+  @memoized_property
   def lib(self):
-    """Load and return the `libgraph` module."""
-    binary = self._binary_util.select_binary(self._supportdir,
-                                              self._version,
-                                              'native-engine')
-    return _FFI.dlopen(binary)
+    """Load and return the native engine module."""
+    return self.ffi.dlopen(self.binary)
+
+  @memoized_property
+  def ffi(self):
```

Comment:
I'm a little confused about the use of `_ffi` vs `ffi`. Why do some methods use one vs the other?

It looks like `ffi` is the ffi for the loaded externs while `_ffi` is the generic cffi interface, which only allows for calling new, new_handle and from_handle. Is that right?